docs/libertyDev.md

```diff
@@ -126,8 +126,13 @@ For the current technology preview, the following limitations apply.
 
 - Dockerfile limitations:
   - The Dockerfile must copy only one .war file for the application.  Other application archive formats or multiple .war files are not supported.
-  - Hot deployment is only supported for individual configuration files that are specified as the source in the Dockerfile's COPY commands. Hot deployment is not supported for COPY commands with variable substitution, wildcard characters, spaces in paths, paths relative to WORKDIR, multi-stage builds, or entire directories specified as the source.
+  - Hot deployment is only supported for individual configuration files that are specified as the source in the Dockerfile's COPY commands. Hot deployment is not supported for COPY commands with entire directories specified as the source. In this case the image it will be rebuilt and the container restarted (regular deployment).
   - Hot deployment is not supported for files or directories copied to the image with the ADD command. When you change any of the files or directories that you ADDed to the image it will be rebuilt and the container restarted.
+  - Certain dockerfile features are not supported by dev mode. In these cases the files are not tracked so if you change these files you must restart the container manually. Type 'r' and press Enter to restart the container.
+   - variable substitution used in the COPY or ADD command e.g. `$PROJECT/config`
+   - wildcards used in the COPY or ADD command e.g. `src/main/liberty/config/*`
+   - paths relative to WORKDIR e.g. `ADD test.txt relativeDir/`
+   - multi-stage builds e.g. `COPY --from=<name>`
```

Comment:
It seems there are now 4 distinct points that we are trying to convey:
1. For individual files specified, we support hot deployment.
2. For directories, we don't support hot deployment but support automatic rebuild and restart.
3. For ADD commands, we don't support hot deployment but support automatic rebuild and restart.
4. For other Dockerfile features, we don't support hot deployment, we don't support automatic rebuild and restart, but we support manual rebuild and restart.

We should consider how to explain the differences between these points and make them clearer.  Otherwise it is confusing about what is supported or not what the user needs to do in different situations.  If you wish, you can also add a statement to cover https://github.com/OpenLiberty/ci.maven/issues/998.

Reply:
Also I think the section should have a different name.

Reply:
New edit available.